### ISSUE TITLE
Include toJson, fromJson methods

### ISSUE THREAD

**Issue by kazbeksultanov** [a14n/dart-decimal#55]
It would be great to have internal methods toJson, fromJson.

**Comment by a14n** [comment]
Unfortunately there is no standard json representation of a Decimal. So only the developer is able to know if a number `2.34` (or a string `"2.43"`) should be converted to a Decimal or something else.

**Comment by kazbeksultanov** [comment]
I am using Decimal as a primitive type in my models, and when it comes to saving data as a JSON there is always error. 
But is there someway to define toJson, fromJson methods in my particular case? I just want to convert to String and save in JSON like this, and parse from String. What can I do?

**Comment by a14n** [comment]
I'm not sure to understand the problem. Do you have something like the following code :

```dart
class MyClass {
  MyClass(this.id, this.value);
  MyClass.fromJson(Map<String, dynamic> jsonMap)
      : this(
          jsonMap['id'] as String,
          Decimal.parse(jsonMap['value'] as String),
        );
  final String id;
  final Decimal value;
  Map<String, dynamic> toJson() => {'id': id, 'value': value.toString()};
}
```

**Comment by kazbeksultanov** [comment]
This is totally fine when you write that on your own. But I am using generated code for serialization. And I can't change every time the generated code. In generator, it is using jsonEncode and jsonDecode methods on types.

**Comment by a14n** [comment]
What serialization tool are you using? Isn't it possible to register custom type handlers?

**Comment by a14n** [comment]
Closed #55 via #60.
